Creates basic app with core activities

rachelbock opened the pull request:
- Creates the shell for the application.
- Creates the home page activity with 4 links to the 4 activities.
- Creates the Action Bar Fragment which hosts the 4 buttons to access the 4 activities.
- Creates a placeholder for the Projects Activity.
- Creates a placeholder for the Wall Activity.
- Creates a placeholder for the UserInfoActivity.
- Adds images to drawables folder that could be used in the application.
- Creates a DialogFragment to collect basic User Information in User Activity.
- Creates layout files for each of the activities.
- Added description to fragment action on HomePage.
- Added description to fragment action on ProjectsActivity.
- Added description and TODO task to UserActivity.
- Added description to fragment action on WallActivity.
- Added method descriptions to ActionBarFragment.
- Added method descriptions and TODO tasks to DialogFragment.

Commit:
Creates a basic app with core activities

-Creates the shell for the application.
-Creates the home page activity with 4 links to the 4 activities.
-Creates the Action Bar Fragment which hosts the 4 buttons to access the 4 activities.
-Creates a placeholder for the Projects Activity.
-Creates a placeholder for the Wall Activity.
-Creates a placeholder for the UserInfoActivity.
-Adds images to drawables folder that could be used in the application.
-Creates a DialogFragment to collect basic User Information in User Activity.
-Creates layout files for each of the activities.

Commit:
Documentation Updates

-Added description to fragment action on HomePage.
-Added description to fragment action on ProjectsActivity.
-Added description and TODO task to UserActivity.
-Added description to fragment action on WallActivity.
-Added method descriptions to ActionBarFragment.
-Added method descriptions and TODO tasks to DialogFragment.

nonvirtualthunk commented on app/src/main/java/com/rage/clamber/Activities/HomePage.java:
```diff
@@ -0,0 +1,24 @@
+import com.rage.clamber.Fragments.ActionBarFragment;
+import com.rage.clamber.R;
+
+public class HomePage extends AppCompatActivity {
```
It's probably a good idea to add a javadoc comment on the class itself as well, describing overall what it's for and what it's expected to do at a high level.

rachelbock replied:
Good idea - I will do that!

nonvirtualthunk commented on app/src/main/java/com/rage/clamber/Fragments/ActionBarFragment.java:
```diff
@@ -0,0 +1,86 @@
+     */
+    public static ActionBarFragment newInstance () {
+        ActionBarFragment fragment = new ActionBarFragment();
+        Bundle args = new Bundle();
```
do you need to create the args bundle, even if the fragment doesn't need any args?

rachelbock replied:
It doesn't look like I would ever need to pass in arguments so I will remove this.

nonvirtualthunk commented on app/src/main/java/com/rage/clamber/Fragments/UserInfoDialogFragment.java:
```diff
@@ -0,0 +1,70 @@
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+
+        View rootView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_user_info_dialog, null);
```
I don't think this is wrong, just wondering, what does the `null` as a second argument to `inflate()` indicate?

rachelbock replied:
It is the parent ViewGroup which is an optional parameter. It is not necessary in this instance as there is no parent.

Commit:
Addition of Class Descriptions

- Added description to each of the Activities and Fragments.
- Removed bundled arguments from ActionBarFragment.

nonvirtualthunk commented:
looks good, merging
